src/Compilers/CSharp/Portable/Binder/Binder_Symbols.cs

```diff
@@ -991,9 +996,14 @@ private bool ShouldCheckConstraints
 
         internal NamedTypeSymbol GetSpecialType(SpecialType typeId, DiagnosticBag diagnostics, CSharpSyntaxNode node)
         {
+            return GetSpecialType(typeId, diagnostics, node.Location);
```

Comment:
@AlekseyTs curious: have we considered a non-allocating solution that allows us to avoid duplicated code?  Essentially something like the following:

``` csharp
struct LocationSource
{
  private readonly Location _location;
  private readonly CSharpSyntaxNode _node;

  public Location Location => _location ?? _node.Location;
 
  public static implicit operator LocationSource(Location location) { ... } 
  public static implicit operator LocationSource(CSharpSyntaxNode node) { ... } 
}
```

Then in any place we would otherwise create an overload we use `LocationSource` instead?